File: leet-code/reverseNumber.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  const max = Math.pow(2, 31) - 1
  const min = -Math.pow(2, 31)
  let rn = 0
  while (x) {
    rn = rn * 10 + (x % 10)
    x = parseInt(x / 10)
  }
  return rn > max || rn < min ? 0 : rn
}

console.log(reverse(-123))

File: leet-code/reverseNumber.go
package main

import "math"

func reverse(x int) int {
	max, min := math.Pow(2, 31)-1, -math.Pow(2, 31)
	res := 0
	for x != 0 {
		res = res*10 + x%10
		x = x / 10
	}
	fres := float64(res)
	if fres > max || fres < min {
		return 0
	}
	return res
}

